sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,179 @@
+    >>> J = PrismaticJoint('J', W, B, coordinates=q, speeds=u)
+    >>> W.apply_force(c*u*W.x, reaction_body=B)
+    >>> W.apply_force(k*q*W.x, reaction_body=B)
+    >>> method = JointsMethod(W, J)
+    >>> (fr, frstar) = method.form_eoms()
+    >>> MM = method.mass_matrix
+    >>> forcing = method.forcing
+    >>> rhs = MM.inv() * forcing
+    >>> rhs
+    Matrix([[(-c*u(t) - k*q(t))/B_mass]])
+
+    """
+
+    def __init__(self, body, *joints):
+        if isinstance(body, Body):
+            self.frame = body.frame
+        else:
+            self.frame = body
+
+        self._joints = joints
+        self._bodylist = self._generate_bodylist()
+        self._loadlist = self._generate_loadlist()
+        self._q = self._generate_q()
+        self._u = self._generate_u()
+        self._kdes = self._generate_kdes()
+
+    @property
+    def bodylist(self):
+        return self._bodylist
+
+    @property
+    def loadlist(self):
+        return self._loadlist
+
+    @property
+    def q(self):
+        return self._q
+
+    @property
+    def u(self):
+        return self._u
+
+    @property
+    def kdes(self):
+        return self._kdes
+
+    @property
+    def forcing_full(self):
+        return self.method.forcing_full
+
+    @property
+    def mass_matrix_full(self):
+        return self.method.mass_matrix_full
+
+    @property
+    def mass_matrix(self):
+        return self.method.mass_matrix
+
+    @property
+    def forcing(self):
+        return self.method.forcing
+
+    @property
+    def method(self):
+        return self._method
+
+    def _generate_bodylist(self):
+        bodies = []
+        for joint in self._joints:
+            if joint.child not in bodies:
+                bodies.append(joint.child)
+            if joint.parent not in bodies:
+                bodies.append(joint.parent)
+        return bodies
+
+    def _generate_loadlist(self):
+        load_list = []
+        for body in self.bodylist:
+            load_list.extend(body.loads)
+        return load_list
+
+    def _generate_q(self):
+        q_ind = []
+        for joint in self._joints:
+            q_ind.extend(joint.coordinates)
+        return q_ind
+
+    def _generate_u(self):
+        u_ind = []
+        for joint in self._joints:
+            u_ind.extend(joint.speeds)
+        return u_ind
+
+    def _generate_kdes(self):
+        kd_ind = []
+        for joint in self._joints:
+            kd_ind.extend(joint.kdes)
+        return kd_ind
+
+    def _form_kanes_equations(self):
+        self._method = KanesMethod(self.frame, q_ind=self.q,
+                                u_ind=self.u, kd_eqs=self.kdes)
+        if self.loadlist == []:
+           fr, frstar = self.method.kanes_equations(self.bodylist)
+           return fr, frstar
+        fr, frstar = self.method.kanes_equations(self.bodylist, self.loadlist)
+        return fr, frstar
+
+    def _form_lagranges_equations(self):
+        pass
+
+    def form_eoms(self, method='kane'):
```

Comment:
Why not pass in `KanesMethod` instead of a string? Then you don't have to add new strings for new methods. Eventually we can get the `*Methods` consistent so things work with duck typing.

Reply:
Since implementation of methods is different , strings seem better to me.

Reply:
The ideal future is that the `*Methods` implementations have a common set of attributes and methods. For the two existing methods, there is only a little bit of work to make those adjustments now (at least for unconstrained systems).